packages/core/content-type-builder/admin/src/components/DataManagerProvider/utils/tests/retrieveNestedComponents.test.ts

```diff
@@ -49,7 +49,83 @@ describe('CONTENT TYPE BUILDER | COMPONENTS | DataManagerProvider | utils | retr
       },
     };
 
-    const expected = ['default.dish'];
+    const expected = [
+      {
+        component: 'default.dish',
+        parentCompoUid: ['default.closingperiod'],
+      },
+    ];
+
+    expect(retrieveNestedComponents(components)).toEqual(expected);
+  });
+
+  it('should return both parents', () => {
+    const components = {
+      'default.closingperiod': {
+        uid: 'default.closingperiod',
+        category: 'default',
+        apiId: 'closingperiod',
+        schema: {
+          icon: 'angry',
+          name: 'closingperiod',
+          description: '',
+          collectionName: 'components_closingperiods',
+          attributes: [
+            { type: 'string', name: 'label' },
+            { type: 'date', required: true, name: 'start_date' },
+            { type: 'date', required: true, name: 'end_date' },
+            { type: 'media', multiple: false, required: false, name: 'media' },
+            { component: 'default.dish', type: 'component', name: 'dish' },
+          ],
+        },
+      },
+      'default.dish': {
+        uid: 'default.dish',
+        category: 'default',
+        apiId: 'dish',
+        schema: {
+          icon: 'address-book',
+          name: 'dish',
+          description: '',
+          collectionName: 'components_dishes',
+          attributes: [
+            { type: 'string', required: false, default: 'My super dish', name: 'name' },
+            { type: 'text', name: 'description' },
+            { type: 'float', name: 'price' },
+            { type: 'media', multiple: false, required: false, name: 'picture' },
+            { type: 'richtext', name: 'very_long_description' },
+            {
+              type: 'relation',
+              relation: 'oneToOne',
+              target: 'api::category.category',
+              targetAttribute: null,
+              private: false,
+              name: 'categories',
+            },
+          ],
+        },
+      },
+
+      'default.openingperiod': {
+        uid: 'default.openingperiod',
+        category: 'default',
+        apiId: 'openingperiod',
+        schema: {
+          icon: 'angry',
+          name: 'openingperiod',
+          description: '',
+          collectionName: 'components_openingperiods',
+          attributes: [{ component: 'default.dish', type: 'component', name: 'dish' }],
+        },
+      },
+    };
+
+    const expected = [
+      {
+        component: 'default.dish',
+        parentCompoUid: ['default.closingperiod', 'default.openingperiod'],
```

Comment:
I find this confusing, why is `parentCompoUID` an array if it's a UID?

Reply:
it's multiple component UIDs, a component can have multiple different parents at the same level

Reply:
I think you should write an explanation and maybe change the name to be clearer, the way you've written it at the moment gives the illusion it's singular or nested i.e. dish is inside openingperiod, which is inside closingperiod 🤔